### PR TITLE
Refactor DockerImagesMetadata

### PR DESCRIPTION
**Pull request by GuyAfik**
## Description
* Refactor the `DockerImagesMetadata` to use the `DockerImage` class for parsing.
* Use the `JsonFile` class to read from github to read the dockerfiles metadata

[nightly:](https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/pipelines/826754 )

### CODE REVIEW

File: demisto_sdk/commands/common/docker_images_metadata.py [hunk 1]
```diff
@@ -72,21 +75,32 @@ def __get_metadata_value(
             docker_image (str): the docker image from the script/integration yml
             docker_metadata_key (str): the key in the DockerImageTagMetadata class
         """
-        try:
-            # if we were not able to load the file
-            if not self.docker_images:
+        if not self.docker_images:
+            return None
+
+        if isinstance(docker_image, str):
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/common/docker_images_metadata.py:81]: not used?

File: demisto_sdk/commands/common/files/file.py [hunk 2]
```diff
@@ -413,6 +425,7 @@ def read_from_http_request(
             timeout: timeout for the request
             handler: whether a custom handler is required, if not takes the default.
             clear_cache: whether to clear cache
+            encoding: whether to apply any custom encoding
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/common/files/file.py:428]: sounds like a boolean. Maybe "the encoding to apply"

**Review by JudahSchwartz — APPROVED** [review]

File: demisto_sdk/commands/common/docker_images_metadata.py (hunk 1 above)
Comment (GuyAfik) [reply demisto_sdk/commands/common/docker_images_metadata.py:81]: wdym?

File: demisto_sdk/commands/common/files/file.py (hunk 2 above)
Comment (GuyAfik) [reply demisto_sdk/commands/common/files/file.py:428]: done

**Pull request merged**